File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        max_sum = 0
        min_len = 10000000000
        l = 0

        for r in range(len(nums)):
            max_sum += nums[r]

            while max_sum >= target:
                min_len = min(min_len, r-l+1)
                max_sum -= nums[l]
                l += 1

        if min_len == 10000000000:
            return 0
        else:
            return min_len
        

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int min = Integer.MAX_VALUE;
        int j = 0, sum = 0, i = 0;
        while(i < nums.length) {
            sum += nums[i];
            while(sum >= target) {
                min = Math.min(min, i - j + 1);
                sum -= nums[j++];
            }
            i++;
        }
        if(min == 2147483647) return 0;
        return min;
    }
}